src/transformations.jl

```diff
@@ -0,0 +1,208 @@
+    d, n = size(X)
+    n >= 2 || error("X must contain at least two columns.")
+
+    m, s = mean_and_std(X, 2)
+
+    return ZScoreTransform(d, (center ? vec(m) : zeros(T, 0)),
+                              (scale ? vec(s) : zeros(T, 0)))
+end
+
+function transform!(y::AbstractVecOrMat{YT}, t::ZScoreTransform, x::AbstractVecOrMat{XT}) where {YT<:Real,XT<:Real}
+    d = t.dim
+    size(x,1) == size(y,1) == d || throw(DimensionMismatch("Inconsistent dimensions."))
+    n = size(y,2)
+    size(x,2) == n || throw(DimensionMismatch("Inconsistent dimensions."))
+
+    m = t.mean
+    s = t.scale
+
+    if isempty(m)
+        if isempty(s)
+            if x !== y
+                copyto!(y, x)
+            end
+        else
+            broadcast!(/, y, x, s)
+        end
+    else
+        if isempty(s)
+            broadcast!(-, y, x, m)
+        else
+            broadcast!((x,m,s)->(x-m)/s, y, x, m, s)
+        end
+    end
+    return y
+end
+
+function reconstruct!(x::AbstractVecOrMat{YT}, t::ZScoreTransform, y::AbstractVecOrMat{XT}) where {YT<:Real,XT<:Real}
+    d = t.dim
+    size(x,1) == size(y,1) == d || throw(DimensionMismatch("Inconsistent dimensions."))
+    n = size(y,2)
+    size(x,2) == n || throw(DimensionMismatch("Inconsistent dimensions."))
+
+    m = t.mean
+    s = t.scale
+
+    if isempty(m)
+        if isempty(s)
+            if y !== x
+                copyto!(x, y)
+            end
+        else
+            broadcast!(*, x, y, s)
+        end
+    else
+        if isempty(s)
+            broadcast!(+, x, y, m)
+        else
+            broadcast!((y,m,s)->y*s+m, x, y, m, s)
+        end
+    end
+    return x
+end
+
+# Unit transformation
+struct UnitRangeTransform{T<:Real}  <: DataTransform
+    dim::Int
+    unit::Bool
+    min::Vector{T}
+    scale::Vector{T}
+
+    function UnitRangeTransform(d::Int, unit::Bool, min::Vector{T}, max::Vector{T}) where T
+        lenmin = length(min)
+        lenmax = length(max)
+        lenmin == d || lenmin == 0 || throw(DimensionMismatch("Inconsistent dimensions."))
+        lenmax == d || lenmax == 0 || throw(DimensionMismatch("Inconsistent dimensions."))
+        new{T}(d, unit, min, max)
+    end
+end
+
+function Base.getproperty(t::UnitRangeTransform, p::Symbol)
+    if p == :indim || p == :outdim
+        return t.dim
+    else
+        return getfield(t, p)
+    end
+end
+
+# fit a unit transform
+function fit(::Type{UnitRangeTransform}, X::AbstractArray{T,2}; unit::Bool=true) where T<:Real
+    d, n = size(X)
+
+    tmin = zeros(T, d)
+    tmax = zeros(T, d)
+    copyto!(tmin, X[:, 1])
+    copyto!(tmax, X[:, 1])
+    for j = 2:n
+        @inbounds for i = 1:d
+            if X[i, j] < tmin[i]
+                tmin[i] = X[i, j]
+            elseif X[i, j] > tmax[i]
+                tmax[i] = X[i, j]
+            end
+        end
+    end
+    for i = 1:d
+        @inbounds tmax[i] = 1.0 / (tmax[i] - tmin[i])
```

Comment:
```suggestion
        @inbounds tmax[i] = 1 / (tmax[i] - tmin[i])
```
(This can help preserving the input type.)